src/software/proto/message_translation/proto_creator_primitive_visitor.cpp

```diff
@@ -23,159 +23,164 @@ PrimitiveMsg ProtoCreatorPrimitiveVisitor::createPrimitiveMsg(const Primitive &p
-    prim->set_parameter2(
+    params->set_parameter2(
         static_cast<float>(chip_primitive.getChipOrigin().y() * MILLIMETERS_PER_METER));
-    prim->set_parameter3(static_cast<float>(
+    params->set_parameter3(static_cast<float>(
         chip_primitive.getChipDirection().toRadians() * CENTIRADIANS_PER_RADIAN));
-    prim->set_parameter4(
+    params->set_parameter4(
         static_cast<float>(chip_primitive.getChipDistance() * MILLIMETERS_PER_METER));
-    prim->set_extra_bits(static_cast<uint8_t>(2 | 1));
-    prim->set_slow(false);
+    params->set_extra_bits(static_cast<uint8_t>(2 | 1));
+    params->set_slow(false);
+
+    prim = PrimitiveMsg();
+    prim->set_allocated_shoot(params);
 }
 
 void ProtoCreatorPrimitiveVisitor::visit(
     const DirectVelocityPrimitive &direct_velocity_primitive)
 {
-    prim = PrimitiveMsg();
-    prim->set_prim_type(
-        PrimitiveMsg::PrimitiveType::PrimitiveMsg_PrimitiveType_DIRECT_VELOCITY);
-    prim->set_parameter1(static_cast<float>(direct_velocity_primitive.getXVelocity() *
-                                            MILLIMETERS_PER_METER));
-    prim->set_parameter2(static_cast<float>(direct_velocity_primitive.getYVelocity() *
-                                            MILLIMETERS_PER_METER));
-    prim->set_parameter3(static_cast<float>(
+    PrimitiveParamsMsg *params = new PrimitiveParamsMsg();
+    params->set_parameter1(static_cast<float>(direct_velocity_primitive.getXVelocity() *
+                                              MILLIMETERS_PER_METER));
+    params->set_parameter2(static_cast<float>(direct_velocity_primitive.getYVelocity() *
+                                              MILLIMETERS_PER_METER));
+    params->set_parameter3(static_cast<float>(
         direct_velocity_primitive.getAngularVelocity() * CENTIRADIANS_PER_RADIAN));
-    prim->set_parameter4(static_cast<float>(0));
-    prim->set_extra_bits(static_cast<uint8_t>(direct_velocity_primitive.getDribblerRpm() *
-                                              DRIBBLER_RPM_TO_RADIO_CONVERSION_FACTOR));
-    prim->set_slow(false);
+    params->set_parameter4(static_cast<float>(0));
+    params->set_extra_bits(
+        static_cast<uint8_t>(direct_velocity_primitive.getDribblerRpm() *
+                             DRIBBLER_RPM_TO_RADIO_CONVERSION_FACTOR));
+    params->set_slow(false);
+
+    prim = PrimitiveMsg();
+    prim->set_allocated_direct_velocity(params);
 }
 
 void ProtoCreatorPrimitiveVisitor::visit(
     const DirectWheelsPrimitive &direct_wheels_primitive)
 {
+    PrimitiveParamsMsg *params = new PrimitiveParamsMsg();
+    params->set_parameter1(static_cast<float>(direct_wheels_primitive.getWheel0Power()));
+    params->set_parameter2(static_cast<float>(direct_wheels_primitive.getWheel1Power()));
+    params->set_parameter3(static_cast<float>(direct_wheels_primitive.getWheel2Power()));
+    params->set_parameter4(static_cast<float>(direct_wheels_primitive.getWheel3Power()));
```

Comment:
this should use frontleft/frontright/... naming instead of wheel0/1/2/3

Reply:
I agree, but I think that updating the `DirectWheelsPrimitive` getter's isn't in the scope of this PR? 

There are several updates required to that class, including using some more representative value for what "power" is. I've filed a ticket for this: https://github.com/UBC-Thunderbots/Software/issues/1590